codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java

```diff
@@ -102,7 +113,12 @@ public boolean generateEndpointBasedParams() {
     }
 
     public boolean includeParam(String name) {
-        return allowedEndpointAuthSchemeParams.contains(name);
+        if (allowedEndpointAuthSchemeParamsConfigured) {
+            return allowedEndpointAuthSchemeParams.contains(name);
+        }
+        // If no explicit allowed endpoint auth scheme params are configured then by default we include all of them except the
+        // ones already defined by default.
+        return !DEFAULT_AUTH_SCHEME_PARAMS.contains(name.toLowerCase(Locale.US));
```

Comment:
I was playing around with this PR for generating S3 and noticed a bug. DefaultS3AuthSchemeProvider.resolveAuthScheme() when building endpointParameters, is missing adding the `region` parameter because of this logic.

Reply:
Good catch, we need separated logic for when to include it for the provider case, I will push a commit to fix this, thanks.

Reply:
Pushed an update.